pkg/maps/lbmap/ipv4.go

```diff
@@ -40,6 +41,36 @@ var (
 
 			return svcKey.ToNetwork(), svcVal.ToNetwork(), nil
 		}).WithCache()
+	Service4MapV2 = bpf.NewMap("cilium_lb4_services_v2",
+		bpf.MapTypeHash,
+		int(unsafe.Sizeof(Service4KeyV2{})),
+		int(unsafe.Sizeof(Service4ValueV2{})),
+		MaxEntries,
+		0, 0,
+		func(key []byte, value []byte) (bpf.MapKey, bpf.MapValue, error) {
+			svcKey, svcVal := Service4KeyV2{}, Service4ValueV2{}
+
+			if err := bpf.ConvertKeyValue(key, value, &svcKey, &svcVal); err != nil {
+				return nil, nil, err
+			}
+
+			return svcKey.ToNetwork(), svcVal.ToNetwork(), nil
```

Comment:
Good spot and good question. I was (blindly) following the existing code. `ToNetwork()` it will just do the swap without checking whether it has been swapped before, so it has the same effect as `ToHost()`. I'm adding to my TODO to fix this in separate PR.